packages/block-library/src/media-text/edit.js

```diff
@@ -39,17 +44,47 @@ const TEMPLATE = [
 ];
 
 class MediaTextEdit extends Component {
-	constructor() {
+	constructor( { attributes } ) {
 		super( ...arguments );
 
 		this.onSelectMedia = this.onSelectMedia.bind( this );
 		this.onWidthChange = this.onWidthChange.bind( this );
 		this.commitWidthChange = this.commitWidthChange.bind( this );
+		this.toggleIsEditing = this.toggleIsEditing.bind( this );
+		this.onSelectUrl = this.onSelectUrl.bind( this );
+		this.renderMediaArea = this.renderMediaArea.bind( this );
 		this.state = {
 			mediaWidth: null,
+			isEditing: ! attributes.mediaUrl,
 		};
 	}
 
+	toggleIsEditing() {
+		this.setState( {
+			isEditing: ! this.state.isEditing,
+		} );
+		if ( this.state.isEditing ) {
+			speak( __( 'You are now viewing the image in the image block.' ) );
+		} else {
+			speak( __( 'You are now editing the image in the image block.' ) );
+		}
+	}
+
+	onSelectUrl( newURL ) {
+		const { mediaUrl } = this.props.attributes;
+
+		if ( newURL !== mediaUrl ) {
+			this.props.setAttributes( {
+				mediaUrl: newURL,
+				mediaId: undefined,
+			} );
+		}
+
+		this.setState( {
+			isEditing: false,
```

Comment:
Should this trigger `speak( __( 'You are now viewing the image in the image block.' ) )`?